packages/react-vapor/src/components/refresh/RefreshCallbackWithButton.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from 'react';
+import {connect} from 'react-redux';
+import {IReactVaporState} from '../../ReactVapor';
+import {callIfDefined} from '../../utils/FalsyValuesUtils';
+import {IDispatch} from '../../utils/ReduxUtils';
+import {Button, IButtonProps} from '../button/Button';
+import {RefreshCallBackActions} from './RefeshCallbackActions';
+import {IRefreshCallbackProps} from './RefreshCallback';
+import {RefreshStatus, RefreshStatusSelectors} from './RefreshCallbackReducer';
+
+export interface IRefreshCallbackWithButtonProps
+    extends IRefreshCallbackProps,
+        Partial<ReturnType<typeof mapStateToProps>>,
+        Partial<ReturnType<typeof mapDispatchToProps>> {}
+
+export interface IRefreshCallbackWithButtonConfig {
+    buttonContainerProps?: React.HtmlHTMLAttributes<HTMLDivElement>;
+    button: IButtonProps;
+}
+
+const mapStateToProps = (state: IReactVaporState, ownProps: IRefreshCallbackWithButtonProps) => ({
+    status: RefreshStatusSelectors.getRefreshStatus(state, {id: ownProps.id}),
+});
+
+const mapDispatchToProps = (dispatch: IDispatch, ownProps: IRefreshCallbackWithButtonProps) => ({
+    start: () => dispatch(RefreshCallBackActions.start(ownProps.id)),
+    stop: () => dispatch(RefreshCallBackActions.stop(ownProps.id)),
+});
+
+export const refreshCallbackWithButton = (config: IRefreshCallbackWithButtonConfig) => (
+    Component: React.ComponentClass<IRefreshCallbackWithButtonProps>
+): React.ComponentClass<Partial<IRefreshCallbackWithButtonProps>> => {
+    class ComponentWithButton extends React.PureComponent<IRefreshCallbackWithButtonProps> {
```

Comment:
class ComponentWithButton extends React.PureComponent<IRefreshCallbackWithButtonProps & ReturnType<typeof mapStateToProps> & ReturnType<typeof mapDispatchToProps>> {

Reply:
Updated with a functionnal Component instead of a HOC